src/lib/tabs/tab-group.ts

```diff
@@ -261,6 +263,14 @@ export class MatTabGroup extends _MatTabGroupMixinBase implements AfterContentIn
     });
   }
 
+  /** Clamps the indexToSelect to the bounds of 0 and the tabs length. */
+  private _clampIndexToSelect(): number {
+    // Note the `|| 0`, which ensures that values like NaN can't get through
+    // and which would otherwise throw the component into an infinite loop
+    // (since Math.max(NaN, 0) === NaN).
+    return Math.min(this._tabs.length - 1, Math.max(this._indexToSelect || 0, 0));
```

Comment:
This function feels a little weird since it's taking the `_indexToSelect` from `this`, but then it's just returning it instead of reassigning. Changing it to a parameter would be clearer and would allow it to be used for other things:

```ts
private _clampIndex(index: number): number {
  return Math.min(this._tabs.length - 1, Math.max(index || 0, 0));
}
```

Reply:
I agree with that. **Note**: I allowed `number | null` as parameter type because I wanted to keep that `NaN` comment somewhere. Otherwise we would need to fallback multiple times to `|| 0`.